kedro-datasets/kedro_datasets/api/api_dataset.py

```diff
@@ -131,11 +203,48 @@ def _execute_request(self, session: Session) -> requests.Response:
         return response
 
     def _load(self) -> requests.Response:
-        with sessions.Session() as session:
-            return self._execute_request(session)
+        if self._request_args["method"] == "GET":
+            with sessions.Session() as session:
+                return self._execute_request(session)
+        else:
+            raise DataSetError("Only GET method is supported for load")
```

Comment:
Can be refactor into this, `else` isn't needed.

```suggestion
        if self._request_args["method"] != "GET":
            raise DataSetError("Only GET method is supported for load")
        with sessions.Session() as session:
            return self._execute_request(session)
```